### PR TITLE
Allow stdin/stdout instead of files for source and object.

### PR DESCRIPTION
**Pull request by epi**

### CODE REVIEW

**Review by pfusik — APPROVED** [review]
LGTM

File: source/app.d [hunk 1]
```diff
@@ -2796,28 +2797,36 @@ void assemblyLine() {
 }
 
 void assemblyFile(string filename) {
-	filename = filename.defaultExtension("asx");
-	if (getOption('p'))
-		filename = absolutePath(filename);
-	File stream = openInputFile(filename);
-	scope (exit) stream.close();
+	immutable(ubyte)[] source = sourceFiles.require(filename, {
```
Comment (pfusik) [inline source/app.d:2800]: https://app.travis-ci.com/github/pfusik/xasm/jobs/589544515

    source/app.d:2800:41: error: no property 'require' for type 'immutable(ubyte)[][string]'
      immutable(ubyte)[] source = sourceFiles.require(filename, {

Comment (pfusik) [reply source/app.d:2800]: https://travis-ci.community/t/could-someone-please-upgrade-the-gdc-travis-use/4533 ?

Comment (epi) [reply source/app.d:2800]: Yes. 4.8.5 is older than Vasco. It's not unreasonable to expect at least GDC 10 nowadays. Probably replacing Travis with some other CI will be more practical than waiting for Travis update.

Comment (pfusik) [reply source/app.d:2800]: Added https://github.com/pfusik/xasm/blob/master/.github/workflows/test.yml
It fails.

Comment (pfusik) [reply source/app.d:2800]: Works now. GDC is not supported.

Comment (epi) [reply source/app.d:2800]: Still better than travis

**Pull request merged**